lib/database_cleaner/active_record/base.rb

```diff
@@ -15,7 +15,11 @@ def self.config_file_location
 
     class Base < DatabaseCleaner::Strategy
       def self.migration_table_name
-        ::ActiveRecord::SchemaMigration.table_name
+        if Gem::Version.new("6.0.0") <= ::ActiveRecord.version
+          ::ActiveRecord::Base.connection.schema_migration.table_name
```

Comment:
I think this should be `connection_class.connection.schema_migration.table_name` to work with multiple databases (as noted on the upstream Rails commit why this was changed), not hardcoded to `ActiveRecord::Base`.

Reply:
hmm..
Since rails 6.0.0 `ActiveRecord::SchemaMigration.table_name` uses `ActiveRecord::Base.table_name_prefix `, `ActiveRecord::Base.table_name`, ` ActiveRecord::Base. table_name_suffix`,
So I think that  the state of the connection is irrelevant to the result of `ActiveRecord::Base.connection.schema_migration.table_name`.

https://github.com/rails/rails/blob/6-0-stable/activerecord/lib/active_record/schema_migration.rb#L21
https://github.com/rails/rails/blob/6-1-stable/activerecord/lib/active_record/schema_migration.rb#L21
https://github.com/rails/rails/blob/7-0-stable/activerecord/lib/active_record/schema_migration.rb#L17
https://github.com/rails/rails/pull/45908/files#diff-353f4f101b1bb04938d761b6e29c88e0f147084e9e276854a77ae5d9a9a3b2a8